gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsStorageStatistics.java

```diff
@@ -276,7 +333,27 @@ synchronized void addSample(long val) {
         return 0;
       }
 
-      return sum / sample;
+      return sum / Math.max(1, sample); // to protect against the race with reset().
     }
+
+    public void reset() {
+      sum = 0;
+      sample = 0;
+    }
+  }
+
+  @Override
+  public String toString() {
```

Comment:
have we added this for testing?

Reply:
No. When the plugin shuts down, it will log all the metrics. This is added for that. Going with the toString implementation of the statistics so that we do not incur the cost if the log is disabled for plugin.